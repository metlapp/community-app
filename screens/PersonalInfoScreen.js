import React, { useContext, useEffect } from "react";
import AccountModal from "../components/AccountModal";
import { Appbar, Button } from "react-native-paper";
import AuthContext from "../auth/Context";
import authStorage from "../auth/Storage";
import { SafeAreaView, StyleSheet, View } from "react-native";
import ListItemWithHeader from "../components/ListItemWithHeader";

const PersonalInfoScreen = () => {
  //determines what component the modal will render when a button is clicked

  const [form, setForm] = React.useState("");
  const [visible, setVisible] = React.useState(false);
  const showModal = () => setVisible(true);
  const hideModal = () => setVisible(false);
  const authContext = useContext(AuthContext);

  const handleLogout = () => {
    //Both sets the user to null and removes the user from async storage
    authContext.setUser(null);
    authStorage.removeUser();
  };
  return (
    <SafeAreaView>
      {/* Opens model and takes in the visible state with what component to render */}
      {/* When ListItem is clicked it changes the state of the visibility and tells the modal what component to render
      inside the modal */}
      <AccountModal visibility={visible} hidemodal={hideModal} form={form} />

        <ListItemWithHeader header="Name" content={authContext.user.first_name}
                  onPress={() => {
                    setForm("Name");
                    showModal();
                  }}/>
        <ListItemWithHeader header="Email" content={authContext.user.email}
                  onPress={() => {
                    setForm("Email");
                    showModal();
                  }}/>
    </SafeAreaView>
  );
};

export default PersonalInfoScreen;
