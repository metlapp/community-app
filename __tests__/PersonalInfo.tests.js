import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import PersonalInfoScreen from "../screens/PersonalInfoScreen";
import {render} from "@testing-library/react-native";
import AuthContext from "../auth/Context";
import {Provider as PaperProvider} from "react-native-paper";

// This test just uses Jest snapshot testing
it("renders correctly, test using Jest", () => {
    const user = {email: "email@email.com"};
    const setUser = jest.fn();
    renderer.create(
        <AuthContext.Provider value={{user, setUser}}>
            <PaperProvider>
                <PersonalInfoScreen/>
            </PaperProvider>
        </AuthContext.Provider>
    );
});

// Using Jest + Enzyme
describe("<PersonalInfoScreen />", () => {
    it("matches the snapshot", () => {
        const user = {email: "email@email.com"};
        const setUser = jest.fn();
        const tree = renderer.create(
            <AuthContext.Provider value={{user, setUser}}>
                <PaperProvider>
                    <PersonalInfoScreen/>
                </PaperProvider>
            </AuthContext.Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Should have correct menu items", () => {
        const user = {email: "email@email.com"};
        const setUser = jest.fn();
        const {getByText} = render(
            <AuthContext.Provider value={{user, setUser}}>
                <PaperProvider>
                    <PersonalInfoScreen/>
                </PaperProvider>
            </AuthContext.Provider>
        );
        expect(getByText('Name')).toBeTruthy();
        expect(getByText('Email')).toBeTruthy();
    });
});
