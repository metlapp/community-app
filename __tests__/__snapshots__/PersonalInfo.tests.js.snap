// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalInfoScreen /> matches the snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "paddingBottom": 5,
                  },
                ],
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "paddingBottom": 5,
                  },
                ],
              ],
            ]
          }
        >
          Name
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Object {
                "color": "#606060",
                "fontSize": 14,
                "opacity": 0.87,
              },
            ]
          }
        />
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "paddingBottom": 5,
                  },
                ],
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Object {
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "paddingBottom": 5,
                  },
                ],
              ],
            ]
          }
        >
          Email
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Object {
                "color": "#606060",
                "fontSize": 14,
                "opacity": 0.87,
              },
            ]
          }
        >
          email@email.com
        </Text>
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
